src/v/security/role.h

```diff
@@ -0,0 +1,86 @@
+/*
+ * Copyright 2024 Redpanda Data, Inc.
+ *
+ * Licensed as a Redpanda Enterprise file under the Redpanda Community
+ * License (the "License"); you may not use this file except in compliance with
+ * the License. You may obtain a copy of the License at
+ *
+ * https://github.com/redpanda-data/redpanda/blob/master/licenses/rcl.md
+ */
+
+#pragma once
+
+#include "absl/container/flat_hash_set.h"
+#include "security/types.h"
+#include "serde/envelope.h"
+
+#include <seastar/core/sstring.hh>
+
+#include <iosfwd>
+
+namespace security {
+
+enum class role_member_type {
+    user = 0,
+};
+
+std::ostream& operator<<(std::ostream&, role_member_type);
+
+class role_member
+  : public serde::
+      envelope<role_member, serde::version<0>, serde::compat_version<0>> {
+public:
+    // NOTE(oren): Default constructor is for serde tag_invoke only.
+    // A default constructed `role_member` is not meaningful and you shouldn't
+    // construct them.
+    role_member() = default;
+    role_member(role_member_type type, ss::sstring name)
+      : _type(type)
+      , _name(std::move(name)) {}
+
+    std::string_view name() const { return _name; }
```

Comment:
generally it would be preferred to return `const string&` here. the caller can then decide to keep the reference `auto& = ` or make a copy `auto =`. with a string_view, both forms result in a reference that is tied to the lifetime of the role_member instance. i guess using `_view` suffix is also fine.

Reply:
Ah, I remember now - I was trying to write a type constraint that isn't possible anyway and never came back to change this. as long as the return type is _convertible_ to std::string_view, we're good. changed in https://github.com/redpanda-data/redpanda/pull/17447